erp/views.py

```diff
@@ -193,104 +193,121 @@ def _update_search_pager(pager, commune):
     return pager
 
 
-def _clean_search_params(params, *args):
-    return ("" if params.get(arg, "") == "None" else params.get(arg, "") for arg in args)
+def _cleaned_search_params_as_dict(get_parameters):
+    allow_list = ("where", "what", "lat", "lon", "code", "search_type", "postcode")
+    return {
+        k: "" if get_parameters.get(k, "") == "None" else get_parameters.get(k, "")
+        for k, v in get_parameters.items()
+        if k in allow_list
+    }
 
 
 def _parse_location_or_404(lat, lon):
-    if not lat or not lon:
+    if not all([lat, lon]):
```

Comment:
approx 4 to 10 times slower than the previous `or`. It is not a big deal, but maybe better to keep it as it was no ?